packages/vite/src/node/server/index.ts

```diff
@@ -988,6 +989,13 @@ export function resolveServerOptions(
 
   if (!allowDirs) {
     allowDirs = [searchForWorkspaceRoot(root)]
+    if (process.versions.pnp) {
+      try {
+        const enableGlobalCache = execSync('yarn config get enableGlobalCache', {cwd: root}).trim() === 'true'
+        const yarnCacheDir = execSync(`yarn config get ${enableGlobalCache ? 'globalFolder' : 'cacheFolder'}`, {cwd: root}).trim()
+        allowDirs.push(yarnCacheDir)
+      } catch {}
```

Comment:
I think we should also do a `logger.warn()` in the catch handler to prevent potential hard-to-debug bugs in the future. Assuming if pnp is enabled, this should always be passing.

Btw I also prefer this child_process approach better 👍